behavior/trim_behavior_video.m

```diff
@@ -0,0 +1,124 @@
+%   video lines up with the concatenated Miniscope movie (e.g. as produced
+%   by `concatenateHDF5` or `concatenate_bigtiff`). The `trim` parameter
+%   needs to match the values used in the Miniscope concatenation.
+%
+% Inputs:
+%   plusmaze_source: Text file output from the plus maze
+%   behavior_source: Behavior video (MPEG-4)
+%   trim: Number of frames to drop from the beginning (trim[1]) and
+%         end (trim[2]) of each trial
+%   
+% Optional input:
+%   dropped_table: [N x 2] matrix where each row indicates the number of
+%       dropped frames in a specified trial of the behavior video.
+%
+%       For example,
+%    
+%           dropped_table = [1 7; 22 6; 42 6];
+%
+%       indicates that Trial 1 has 7 dropped frames; Trial 22 has 6 dropped
+%       frames; Trial 42 has 6 dropped frames.
+%
+% Example usage:
+%   dropped_table = load('mouse7_day07_ego-left_dropped_behavior_frames.txt');
+%   trim_behavior_video('mouse7_d07_ego-left.txt', 'mouse7_day07_ego-left.m4v', [15 5], dropped_table)
+%
+
+
+
+% Get the trial frames (beginning and end) according to PlusMaze output.
+%   We regard the PlusMaze trial indices as ground truth!
+%----------------------------------------------------------------------
+orig_frame_indices = double(get_trial_frame_indices(plusmaze_source));
+orig_frame_indices = orig_frame_indices(:,[1 4]); % Keep [Start End]
+
+num_trials = size(orig_frame_indices,1);
+num_frames = orig_frame_indices(num_trials,2); % Very last frame
+fprintf('  PlusMaze output (%s) has %d frames\n', plusmaze_source, num_frames);
+
+% Generate frame indices into the behavior video, with its dropped frames
+%   (Optional parameter specifies a table of dropped frames)
+%----------------------------------------------------------------------
+dropped_frames = zeros(num_trials,1);
+if ~isempty(varargin)
+    dropped_frames_sparse = varargin{1}; % [Trial-number Num-dropped-frames]
+    dropped_frames(dropped_frames_sparse(:,1)) = dropped_frames_sparse(:,2);
+end
+
+dropped_frame_indices = compute_dropped_frame_indices(orig_frame_indices, dropped_frames);
+
+% Read behavior video, and make sure that the dropped trial indices
+%   match up with the behavior video's number of frames
+%----------------------------------------------------------------------
+behavior_video = VideoReader(behavior_source);
+num_behavior_frames = behavior_video.NumberOfFrames;
+fprintf('  Behavior video (%s) has %d frames\n', behavior_source, num_behavior_frames);
+
+assert(num_behavior_frames == dropped_frame_indices(end,2),...
+       '  Dropped frame table is inconsistent with actual number of frames in behavior movie!');
+
+% We don't handle the case where the number of frames in the behavior video
+%   exceeds the frame count indicated by the PlusMaze
+num_actual_dropped_frames = num_frames - num_behavior_frames;
+assert(num_actual_dropped_frames >= 0,...
+       '  Behavior video has more frames (%d) than indicated by PlusMaze output (%d)!',...

```

Comment:
It's correct as written.

Matlab's `assert` makes sure that the first condition (`num_actual_dropped_frames >= 0`) is true. The text is only displayed in the case that the assertion fails (i.e. the case `num_actual_dropped_frames < 0`)